### ISSUE TITLE
Marking backups as "in progress" or "finished"

### ISSUE THREAD

**Issue by abbbi** [tizbac/pmoxs3backuproxy#49]
While working on the sync feature #48 i noticed that for sync the syncjob checks by identifying if an backup is active by using the "Size" flag of an snapshot:

```
       Ok(snapshot_list
            .into_iter()
            .filter_map(|item: SnapshotListItem| {
                let snapshot = item.backup;
                // in-progress backups can't be synced                                                                                  
                if item.size.is_none() {
                    task_log!(
                        worker,
                        "skipping snapshot {} - in-progress backup",
```

while we can "fake" this response for the sync to work, the current implementation has no real way to determine if an
backup is still active.

I think we could either use an "active" flag which is added as the backup starts and cleared as it is finished?

**Comment by tizbac** [comment]
Yes or simply we could keep track of it in the proxy itself, without doing
it in S3 since anyway all running backups are necessarily in memory

Il mar 20 ago 2024, 10:34 Michael Ablassmeier ***@***.***> ha
scritto:

**Comment by tizbac** [comment]
Only edge case it comes to my mind is proxy crashing while doing backup and unable to remove it, a clean way without duplication would be setting the corrupt flag and removing it when done

**Comment by abbbi** [comment]
> Only edge case it comes to my mind is proxy crashing while doing backup and unable to remove it, a clean way without duplication would be setting the corrupt flag and removing it when done

yes. Also, signal handling should be added so it could be shutdown "gracefully", means if it receives kill signal and there are still active sessions, wait until they are gone for X seconds, then shutdown.  

Maybe it shouldnt write the stuff into the backups/ folder in the first place,but something like active/ first, and if the backup finishes move the files to the backups folder (atomic).
